bgpd/bgp_nb_config.c

```diff
@@ -313,14 +641,23 @@ int routing_control_plane_protocols_control_plane_protocol_bgp_global_route_refl
 int routing_control_plane_protocols_control_plane_protocol_bgp_global_route_reflector_no_client_reflect_modify(
 	struct nb_cb_modify_args *args)
 {
-	switch (args->event) {
-	case NB_EV_VALIDATE:
-	case NB_EV_PREPARE:
-	case NB_EV_ABORT:
-	case NB_EV_APPLY:
-		/* TODO: implement me. */
-		break;
-	}
+	if (args->event != NB_EV_APPLY)
+		return NB_OK;
+
+	struct bgp *bgp;
+	int ret;
+
+	bgp = nb_running_get_entry(args->dnode, NULL, true);
+
+	if (yang_dnode_get_bool(args->dnode, NULL))
+		SET_FLAG(bgp->flags, BGP_FLAG_NO_CLIENT_TO_CLIENT);
+	else
+		UNSET_FLAG(bgp->flags, BGP_FLAG_NO_CLIENT_TO_CLIENT);
+
+	ret = bgp_clear_star_soft_out(bgp->name, args->errmsg,
```

Comment:
We need to have a check to see if the flags has flipped only then call bgp_clear_star_soft_out.

Reply:
Isn't northbound callback only invoked when flag is flipped?